game/src/main/scala/ScalatraBootstrap.scala

```diff
@@ -6,23 +6,34 @@ import javax.servlet.ServletContext
 import io.aigar.model.TeamRepository
 
 class ScalatraBootstrap extends LifeCycle {
-  val teamRepository = ScalatraBootstrap.fixedTeamRepository.getOrElse(new TeamRepository(None))
-  val game = new GameThread(fetchTeamIDs)
+  var teamRepository: TeamRepository = null
+  var game: GameThread = null
 
   override def init(context: ServletContext): Unit = {
-    launchGameLoop
+    appInit()
 
     val path = "/api/1"
     context.mount(new LeaderboardController, s"$path/leaderboard/*")
     context.mount(new GameController(game), s"$path/game/*")
   }
 
+  /*
+   * Separated method for testing purposes.
+   */
+  def appInit(teams: Option[TeamRepository] = None): Unit = {
+    teamRepository = teams.getOrElse(new TeamRepository(None))
+    game = new GameThread(fetchTeamIDs)
+
+    launchGameLoop
+  }
+
   private def closeDbConnection {
     teamRepository.closeConnection
   }
 
   override def destroy(context: ServletContext) {
     super.destroy(context)
+    game.running = false
```

Comment:
The add of `running` attribute, is it to prevent problem when killing the app? Is it only to do the loop in a cleaner way or it really changes something?